pkgs/development/libraries/qtkeychain/default.nix

```diff
@@ -1,13 +1,15 @@
 { stdenv, fetchFromGitHub, cmake, qt4 ? null
 , withQt5 ? false, qtbase ? null, qttools ? null
+, darwin ? null
 }:
 
 assert withQt5 -> qtbase != null;
 assert withQt5 -> qttools != null;
+assert stdenv.isDarwin -> darwin != null;
 
 stdenv.mkDerivation rec {
   name = "qtkeychain-${if withQt5 then "qt5" else "qt4"}-${version}";
-  version = "0.7.0";
```

Comment:
Hmm, is 0.7.0 and 0.8.0 the same revision?

Reply:
Right, the unchanged hash is a mistake, apparently.

Reply:
Thing is I tested both versions with `--check` and neither gives a hash mismatch

Reply:
Yea I overlooked changing `sha256`. I'm was seeing some odd behavior though: changing `sha256` controlled the version build, regardless of the value of `rev`, but it seems that a `nix-collect-garbage` may work around this.

1. set correct `rev` and `sha256` (eg `v0.6.0` and `0sbgy1vxabcnn1qi9dfxi6yd0gbbqfgx7sj0c381mp4xay9fxl21`)
1. `nix-build -A qtkeychain` (successfully builds)
1. built version is `0.6.0` (`set(PACKAGE_VERSION "0.6.0")` in `result/lib/cmake/QtKeychain/QtKeychainConfigVersion.cmake`)
1. fudge the `rev` but keep `sha256` (eg `rev -> v0.8.0`)
1. `nix-build -A qtkeychain` (successfully builds)
1. built version is  still`0.6.0`

I'll set the correct hash

Reply:
Fetches are hash-addressed ("fixed-output" in our terminology).  If you have the specified hash in nix store, it will be used without any further checks.

Reply:
Ahhh :) I understand now, thanks!

Reply:
You can force a check with `nix-build '<nixpkgs>' -A hello.src --check` for example.